prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusSinkBuilder.java

```diff
@@ -0,0 +1,246 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.connector.prometheus.sink;
+
+import org.apache.flink.connector.base.sink.AsyncSinkBase;
+import org.apache.flink.connector.base.sink.AsyncSinkBaseBuilder;
+import org.apache.flink.connector.prometheus.sink.errorhandling.SinkWriterErrorHandlingBehaviorConfiguration;
+import org.apache.flink.connector.prometheus.sink.http.PrometheusAsyncHttpClientBuilder;
+import org.apache.flink.connector.prometheus.sink.http.RetryConfiguration;
+import org.apache.flink.connector.prometheus.sink.prometheus.Types;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+/** Builder for Sink implementation. */
+public class PrometheusSinkBuilder
+        extends AsyncSinkBaseBuilder<
+                PrometheusTimeSeries, Types.TimeSeries, PrometheusSinkBuilder> {
+    private static final Logger LOG = LoggerFactory.getLogger(PrometheusSinkBuilder.class);
+
+    private static final int DEFAULT_MAX_BATCH_SIZE_IN_SAMPLES = 500;
+    private static final long DEFAULT_MAX_TIME_IN_BUFFER_MS = 5000;
+    private static final int DEFAULT_MAX_BUFFERED_REQUESTS =
+            1000; // Max nr of requestEntry that will be buffered
+
+    private String prometheusRemoteWriteUrl;
+    private RetryConfiguration retryConfiguration;
+    private Integer socketTimeoutMs;
+    private PrometheusRequestSigner requestSigner = null;
+    private Integer maxBatchSizeInSamples;
+    private Integer maxRecordSizeInSamples;
+    private String httpUserAgent = null;
+    private SinkWriterErrorHandlingBehaviorConfiguration errorHandlingBehaviorConfig = null;
+
+    @Override
+    public AsyncSinkBase<PrometheusTimeSeries, Types.TimeSeries> build() {
+
+        int maxInFlightRequest = 1;
```

Comment:
nit: this could be a static constant. (I understand this is a copy of the real value in the sink itself)